File: courses/CS50/pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void) {

    float change;
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int penny = 1;

    int coin_count = 0;

    do {
        change = get_float("Change owed: ");
    }
    while (change <= 0);

    change = round(change * 100);


    while (change != 0) {
        if (change >= 25) {
            change = change - quarter;
            coin_count++;
        }
        else if (change < 25 && change >= 10) {
            change = change - dime;
            coin_count++;
        }
        else if (change < 10 && change >= 5) {
            change = change - nickel;
            coin_count++;
        }
        else if (change < 5 && change >= 1) {
            change = change - penny;
            coin_count++;
        }
        //printf("%f\n", change);
    }
   printf("%i\n", coin_count);
}

File: courses/CS50/pset6/cash/cash.py
from cs50 import get_float

while True:
    change = get_float("Change owed: ")
    if change > 0:
        break

quarter = 25
dime = 10
nickel = 5
penny = 1

count = 0

change = int(change * 100)

while change != 0:
    if change >= quarter:
        change -= quarter
        count += 1
    elif change >= dime and change < quarter:
        change -= dime
        count += 1
    elif change >= nickel and change < dime:
        change -= nickel
        count += 1
    elif change >= penny and change < nickel:
        change -= penny
        count += 1

print(count)